portal/static/js/utility.js

```diff
@@ -343,6 +343,29 @@ function getUrlParameter(name) {
     var results = regex.exec(location.search);
     return results === null ? "" : decodeURIComponent(results[1]);
 }
+function resetBrowserBackHistory(locationUrl, stateObject, title) {
+    var historyDefined = typeof history !== "undefined" && history.pushState;
+    locationUrl = locationUrl || location.href;
+    if (historyDefined) {
+        history.pushState(stateObject, title, locationUrl);
+    }
+    window.addEventListener("popstate", function(e) {
+        if (historyDefined) {
+            history.pushState(stateObject, title, locationUrl);
+        } else {
+            window.history.forward(1);
+        }
+    });
+}
+function handlePostLogout() {
+    if (typeof sessionStorage === "undefined") {
+        return false;
+    }
+    if (sessionStorage.getItem("logout")) {
+        resetBrowserBackHistory(location.orgin, "logout"); /* global resetBrowserBackHistory */
+        sessionStorage.removeItem("logout");
```

Comment:
If multiple tabs are open, I think this may only clear the history in the first one. It may be more robust to assume they're not logged in if there's no user cached in sessionStorage

Reply:
> to assume they're not logged in if there's no user cached in sessionStorage

hmm, will this touch on other unauthenticated pages as well if we are making that assumption (e.g.navigate away to an unauthenticated page upon logging out)?

Reply:
ah right, we wouldn't want to clear it for **every** unauthenticated page